src/helpers.h

```diff
@@ -21,9 +21,172 @@ along with this program.  If not, see http://www.gnu.org/licenses/
+        throw std::overflow_error("Border too large");
+
+    std::ostringstream sb;
+    sb << "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n";
+    sb << "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\" \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n";
+    sb << "<svg xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\" viewBox=\"0 0 ";
+    sb << (qr.getSize() + border * 2) << " " << (qr.getSize() + border * 2) << "\" stroke=\"none\">\n";
+    sb << "\t<rect width=\"100%\" height=\"100%\" fill=\"#FFFFFF\"/>\n";
+    sb << "\t<path d=\"";
+    for (int y = 0; y < qr.getSize(); y++)
+    {
+        for (int x = 0; x < qr.getSize(); x++)
+        {
+            if (qr.getModule(x, y))
+            {
+                if (x != 0 || y != 0)
+                    sb << " ";
+                sb << "M" << (x + border) << "," << (y + border) << "h1v1h-1z";
+            }
+        }
+    }
+    sb << "\" fill=\"#000000\"/>\n";
+    sb << "</svg>\n";
+    return sb.str();
+}
+
+// Return filename for later deletion
+// Only input filename as "name.ext" and don't include the path
+static std::string writeSvgToDisk(std::string filename, std::string contents)
+{
+    filename = std::filesystem::temp_directory_path().generic_string() + filename;
+    std::ofstream svg(filename, std::ios::out | std::ios::trunc);
+
+    svg.write(contents.c_str(), contents.length());
+
+    svg.flush();
+    svg.close();
+
+    return filename;
+}
+
+ template <typename T>
+struct always_false : std::false_type
+{
+};
+
+template <typename T>
+inline constexpr bool always_false_v = always_false<T>::value;
+
+template <typename T>
+inline std::optional<T> jsonGet(const json& jsonInput, std::string key)
+{
+    if (!jsonInput.contains(key))
+    {
+        return std::nullopt;
+    }
+
+    // Check types first, boolean can match a number
+    if constexpr (std::is_same_v<T, std::string>)
+    {
+        if (!jsonInput[key].is_string())
+        {
+            return std::nullopt;
+        }
+    }
+    else if constexpr (std::is_same_v<T, bool> || std::is_same_v<T, boolean>)
+    {
+        if (!jsonInput[key].is_boolean())
+        {
+            return std::nullopt;
+        }
+    }
+    else if constexpr (std::is_floating_point<T>::value)
+    {
+        if (!jsonInput[key].is_number_float())
+        {
+            return std::nullopt;
+        }
+    }
+    else if constexpr (std::is_signed<T>::value)
+    {
+        if (!jsonInput[key].is_number_unsigned())
+        {
+            return std::nullopt;
+        }
+    }
+    else if constexpr (std::is_unsigned<T>::value)
+    {
+        if (!jsonInput[key].is_number_unsigned())
+        {
+            return std::nullopt;
+        }
+    }
+    else
+    {
+        static_assert(always_false_v<T>, "Trying to extract unsupported type from jsonGet");
+    }
+
+    return jsonInput[key].get<T>();
+}
+
+// Supposedly, there is template magic to do this inside the template above, but VC++ doesn't support it yet?
```

Comment:
Send me whatever error you're facing, this is all absolutely possible and above board